main/lsp/requests/hover.cc

```diff
@@ -14,22 +14,34 @@ namespace sorbet::realmain::lsp {
 string methodInfoString(const core::GlobalState &gs, const core::TypePtr &retType,
                         const core::DispatchResult &dispatchResult,
                         const unique_ptr<core::TypeConstraint> &constraint) {
-    string contents;
+    vector<string> contents;
     auto start = &dispatchResult;
-    ;
+
     while (start != nullptr) {
         auto &component = start->main;
         if (component.method.exists()) {
-            if (!contents.empty()) {
-                contents += "\n";
+            auto method = component.method;
+
+            auto localRetType =
+                getResultType(gs, method.data(gs)->resultType, method, component.receiver, constraint.get());
+            if (localRetType == nullptr || localRetType.isUntyped()) {
+                localRetType = retType;
             }
-            contents = absl::StrCat(
-                contents, prettyTypeForMethod(gs, component.method, component.receiver, retType, constraint.get()));
+
+            contents.push_back(prettySigForMethod(gs, method, component.receiver, localRetType, constraint.get()));
         }
         start = start->secondary.get();
     }
 
-    return contents;
+    if (dispatchResult.main.method.exists()) {
+        if (contents.size() > 1) {
+            contents.push_back(fmt::format("\n-> {}", retType.show(gs)));
+        }
+
+        contents.push_back(prettyDefForMethod(gs, dispatchResult.main.method));
+    }
+
+    return absl::StrJoin(contents, "\n");
```

Comment:
Still need to remove the duplicates from the generated list, but would love some feedback on this as an approach.